src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallNotificationService.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.digital.hmpps.managerecallsapi.service
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.managerecallsapi.documents.DocumentDetail
+import uk.gov.justice.digital.hmpps.managerecallsapi.documents.InputStreamDocumentDetail
+import uk.gov.justice.digital.hmpps.managerecallsapi.documents.PdfDocumentGenerator
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.RecallId
+import java.time.Duration
+
+@Service
+class RecallNotificationService(
+  @Autowired private val revocationOrderService: RevocationOrderService,
+  @Autowired private val recallSummaryService: RecallSummaryService,
+  @Autowired private val pdfDocumentGenerator: PdfDocumentGenerator
+) {
+
+  fun getDocument(recallId: RecallId): Mono<ByteArray> {
+    val docs = mutableListOf<DocumentDetail<out Any>>()
+
+    recallSummaryService.getPdf(recallId).block(Duration.ofSeconds(5))?.let {
```

Comment:
Can we try to refactor this so that we don't block the flux? Happy to pair on it with you.

Reply:
Yeh, @mikedunfordEE and I will do that shortly. But wanted to merge this and then refactor on next PR.

Reply:
Yes we can tidy this up this afternoon

Reply:
cool